bft/bft_interp/src/lib.rs

```diff
@@ -214,7 +224,11 @@ where
             }
         }
         self.head += 1;
-        Ok(())
+        Ok(self.next_instruction())
+    }
+
+    pub fn current_cell(&mut self) -> &mut T {
+        &mut self.cells[self.head]
     }
```

Comment:
I'm not clear why you need this function, using this can cause borrow problems in functions in your impl blocks, because to call it, everything inside the instance must be consistent and unborrowed - I imagine you're not having a problem right now, but in general I try and avoid this kind of function unless it's critical to your API.  Do you need this to be public?  If not, perhaps it need not exist.

If you're not quite sure why I'm saying this is a potential problem; then think more about what it means to be able to call a method on a struct  in terms of the borrows.  If you're still stuck, poke me IRL and I'll try to explain it another way.